src/master/integration/use-cases.md

```diff
@@ -0,0 +1,13 @@
+# Integration between immudb and other data technologies
+
+## Use cases
+
+immudb is designed to work with immutable data, using the relational model or the key/value model. For this reason it’s logical to use it as a primary way to persist the valuable and critical data.
+
+Not all data is valuable or critical. Non-critical data include caches, star schemas that can always be rebuilt, and time series data that will only be preserved for one week or two. This kind of data should not be persisted in immudb: for each of them, more suitable technologies can be used.
+
+While different data sets have different requirements and need to be stored in different technologies, sometimes they need to be used together. That’s the case, for example, when a data warehouse is built from heterogeneous sources. It is natural to move data from immudb to other relational databases to allow for further analyses.
+
+Another case is when data cannot be persisted into immudb from the beginning. For example because an existing application writes it into another DBMS, like MariaDB or PostgreSQL. This may be reliable enough as a first solution, because all major DBMSs provide durability and support high availability solutions. Data that should be stored by an immutable technology can be periodically moved from other DBMSs to immudb, for example every night or every weekend when the production workload is lower.
```

Comment:
I see. Would this works for you? If I misunderstood your comment please let me know.

> Another case is when data cannot be persisted into immudb from the beginning. For example because an existing application writes it into another DBMS, like MariaDB or PostgreSQL. This may be reliable enough as a first solution, because all major DBMSs provide durability and support high availability solutions. Data that should be stored by an immutable technology can be periodically moved from other DBMSs to immudb, for example every night or every weekend when the production workload is lower. This means that the program that reads the original database is responsible for inserting correct data into immudb. In those cases, typically we will also want to audit that program's behaviour, logging its actions into another immudb database.

Reply:
However, if the problem is that the data copy program can be have bugs or malicious behaviours, isn't this true for all programs? In practice, should this mean that we need to log the requests made by the user to the application and the queries it runs against the primary database? But both things could be impossible.

Reply:
The biggest benefit while using an immutable technology is that applications are built on top of mathematically prooven data and transactions. To achive this a foundamental piece of the puzzle is the immudb client sdk that implements the auditor pattern. If it's not possible for clients to read and write with this sdk ( because immudb is a sidecar db as instance ) one or more external auditor(s) are the way to guarantee consistency to whole system.
So it's not needed to log, only to audit.

Reply:
@mmeloni  Thanks for explaining. Would you help me check if I understand when the proof is there and when it isn't?
* Rows inserted via an ODBC connection don't have a proof
* Data inserted via immugw can prove that they've been inserted via immugw and then not altered
* If you could be sure that all data are inserted via `safeset` you could safely read them via `get`; but if data inserted via `set` are read via `safeget`, altered data are not returned
* In all other cases, the proof is there

Reply:
Hi @federico-razzoli,

> Rows inserted via an ODBC connection don't have a proof

true

> Data inserted via immugw can prove that they've been inserted via immugw and then not altered

Immugw provides tamperproof verification so client sending rest payloads will know if immugw was able to verify correctly the sets or gets. Note: Immugw internally uses immuclient SDK. This means that immugw works like an auditor and this is the reason why we recommend to run immudb and immugw on separate machines: status saved by immugw need to be store in a different place in case immudb server part is compromised. 

> If you could be sure that all data are inserted via safeset you could safely read them via get; but if data inserted via set are read via safeget, altered data are not returned

FALSE:
- if immudb is compromised an attacker could returns the right **safeset** proof(client verification wil pass) but a tampered **get** response. Simple **get** doesn't verify consistency and integrity. 
- if data is inserted via **set** in a compromised IMMUDB following **safeSet** will be able to detect the tamper on client side and if **set** payload is manipulated a tamper alert is returned.

Reply:
@mmeloni thank you, now everything is much more clear to me. I've added a paragraph about tamperproof, would you please check that everything is accurate?